### PR TITLE
aws - add kms filter for dyanmodb-table resources

### PR DESCRIPTION
**Pull request by RobMohr**
Added kms filter for dynamodb-table resources by leveraging the KmsRelatedFilter

### CODE REVIEW

**Comment by kapilt** [comment]
Thanks for the pull request, lgtm. I did want to double check on why its listing all kms keys instead of just the ones needed on the alias.

File: tests/test_dynamodb.py [hunk 1]
```diff
@@ -80,6 +80,26 @@ def test_tag_filter(self):
...
+                        "key": "c7n:AliasName",
+                        "value": "^(alias/aws/dynamodb)",
+                        "op": "regex"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```
Comment (kapilt) [inline tests/test_dynamodb.py]: should also add an assertion of the table name attribute to ensure its actually getting the intended resource, the cardinality checks are a weak form of verification that we've been moving away from in favor of an attribute check.

Comment (RobMohr) [reply tests/test_dynamodb.py]: Good point. Added a new assert on the name

**Comment by RobMohr** [comment]
> I did want to double check on why its listing all kms keys instead of just the ones needed on the alias.

I believe this is because the KmsRelatedFilter uses kms-key resource which has list_keys as its enumerator

**Comment by kapilt** [comment]
re why on the enumeration of keys, its because list keys does support filtering

**Pull request merged**